elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -46,63 +54,198 @@
+     * This is the main method which processes the Async Query request, executes the query and updates
+     * values for AsyncQuery and AsyncQueryResult models accordingly.
+     * @return AsyncQueryResult
+     * @throws URISyntaxException
+     * @throws NoHttpResponseException
+     */
     protected AsyncQueryResult processQuery() throws URISyntaxException, NoHttpResponseException {
         UUID requestId = UUID.fromString(queryObj.getRequestId());
 
         ElideResponse response = null;
+        Integer recCount = null;
+
+        Date startTime = new Date();
+
         log.debug("AsyncQuery Object from request: {}", queryObj);
         if (queryObj.getQueryType().equals(QueryType.JSONAPI_V1_0)) {
             MultivaluedMap<String, String> queryParams = getQueryParams(queryObj.getQuery());
             log.debug("Extracted QueryParams from AsyncQuery Object: {}", queryParams);
             response = elide.get(getPath(queryObj.getQuery()), queryParams, user, apiVersion, requestId);
             log.debug("JSONAPI_V1_0 getResponseCode: {}, JSONAPI_V1_0 getBody: {}",
                     response.getResponseCode(), response.getBody());
+
+            if (response.getResponseCode() == 200) {
+                recCount = calculateRecordsJSON(response.getBody());
+            }
         }
         else if (queryObj.getQueryType().equals(QueryType.GRAPHQL_V1_0)) {
             response = runner.run(queryObj.getQuery(), user, requestId);
             log.debug("GRAPHQL_V1_0 getResponseCode: {}, GRAPHQL_V1_0 getBody: {}",
                     response.getResponseCode(), response.getBody());
+
+            if (response.getResponseCode() == 200) {
+                String tableName = getTableNameFromQuery(queryObj.getQuery());
+                recCount = calculateRecordsGRAPHQL(response.getBody(), tableName);
+            }
         }
         if (response == null) {
             throw new NoHttpResponseException("Response for request returned as null");
         }
 
+        long timeElapsed = ((new Date()).getTime() - startTime.getTime());
+        long asyncAfterMilliSeconds = queryObj.getAsyncAfterSeconds() * 1000;
+
         // Create AsyncQueryResult entry for AsyncQuery
 
         queryResultObj = new AsyncQueryResult();
         queryResultObj.setHttpStatus(response.getResponseCode());
-        queryResultObj.setResponseBody(response.getBody());
         queryResultObj.setContentLength(response.getBody().length());
-        queryResultObj.setResultType(ResultType.EMBEDDED);
+        queryResultObj.setRecordCount(recCount);
         queryResultObj.setCompletedOn(new Date());
 
+        if (timeElapsed - asyncAfterMilliSeconds > 0) {
+
+            URL url = null;
+            if (queryObj.getResultFormatType() == ResultFormatType.CSV) {
+                String jsonToCSV = convertJsonToCSV(response.getBody());
+                byte[] temp = jsonToCSV.getBytes();
+                url = resultStorageEngine.storeResults(queryObj.getId(), temp);
```

Comment:
This is common between if and else. This could be moved outside the if else blocks.